helper/pluginutil/version.go

```diff
@@ -0,0 +1,34 @@
+package pluginutil
+
+import (
+	"os"
+
+	gversion "github.com/hashicorp/go-version"
+)
+
+var (
+	// PluginVaultVersionEnv is the ENV name used to pass the version of the
+	// vault server to the plugin
+	PluginVaultVersionEnv = "VAULT_VERSION"
+)
+
+// GRPCSupport returns true if Vault can support GRPC transport, false otherwise
+func GRPCSupport() bool {
+	verString := os.Getenv(PluginVaultVersionEnv)
+
+	if verString != "" && verString != "unknown" {
+		ver, err := gversion.NewVersion(verString)
+		if err != nil {
+			return false
+		}
+
+		constraint, err := gversion.NewConstraint(">= 0.9.2")
+		if err != nil {
+			return false
+		}
+
+		return constraint.Check(ver)
+	}
+
+	return false
```

Comment:
I think we should default to using/supporting gRPC instead of the other way around.

Reply:
In this case if the version is unknown (e.g. from a master build that didn't inherit version from a git tag), it will end up will using netrpc.

Reply:
Actually I see why you're doing this (to maintain backward compatibility in the case PluginVaultVersionEnv is not provided due to older versions of the plugin). Maybe `if verString == ""` should be its own check.

Reply:
Unfortunately we can't because older versions of vault won't be setting the ENV variable. If the variable is empty we can't use gRPC

Reply:
Right so check for `verString != ""` early and short-circuit by returning false. We can then return true here and in the error cases.